### ISSUE TITLE
get irrelevant bogus error message 1st order diff eq

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1516]
Imported from SourceForge on 2024-07-04 18:03:45
Created by **[zmth](https://sourceforge.net/u/zmth/)** on 2021-12-08 06:46:41
Original: https://sourceforge.net/p/maxima/bugs/3899

---

~~~
build_info(version="branch_5_44_base_231_g5c411f69f",timestamp="2021-01-12 23:51:42",host="x86_64-w64-mingw32",lisp_name="SBCL",lisp_version="2.0.0",maxima_userdir="C:/Users/zmth1/maxima",maxima_tempdir="C:/Users/zmth1/AppData/Local/Temp",maxima_objdir="C:/Users/zmth1/maxima/binary/branch_5_44_base_231_g5c411f69f/sbcl/2_0_0",maxima_frontend="wxMaxima",maxima_frontend_version="20.12.2-DevelopmentSnapshot_MSW_OpenMP201511+Locks")build_info(version="branch_5_44_base_231_g5c411f69f",timestamp="2021-01-12 23:51:42",host="x86_64-w64-mingw32",lisp_name="SBCL",lisp_version="2.0.0",maxima_userdir="C:/Users/zmth1/maxima",maxima_tempdir="C:/Users/zmth1/AppData/Local/Temp",maxima_objdir="C:/Users/zmth1/maxima/binary/branch_5_44_base_231_g5c411f69f/sbcl/2_0_0",maxima_frontend="wxMaxima",maxima_frontend_version="20.12.2-DevelopmentSnapshot_MSW_OpenMP201511+Locks")(version="branch_5_44_base_231_g5c411f69f",timestamp="2021-01-12 23:51:42",host="x86_64-w64-mingw32",lisp_name="SBCL",lisp_version="2.0.0",maxima_userdir="C:/Users/zmth1/maxima",maxima_tempdir="C:/Users/zmth1/AppData/Local/Temp",maxima_objdir="C:/Users/zmth1/maxima/binary/branch_5_44_base_231_g5c411f69f/sbcl/2_0_0",maxima_frontend="wxMaxima",maxima_frontend_version="20.12.2-DevelopmentSnapshot_MSW_OpenMP201511+Locks")
~~~
~~~
desolve('diff(f(x),x)+f(x),x)
~~~
get bogus irrelevant error message :
~~~
length: argument cannot be a symbol; found x
 -- an error. To debug this try: debugmode(true);
~~~ 
 when length has nothing to do with  it,irrelevant.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 18:03:47
Created by **[l_butler](https://sourceforge.net/u/l_butler/)** on 2021-12-08 13:20:18
Original: https://sourceforge.net/p/maxima/bugs/3899/#2133

---

- **labels**:  --> desolve, error handling
- Description has changed:

Diff:

~~~~

--- old
+++ new
@@ -1,9 +1,12 @@
+~~~
 build_info(version=&#34;branch_5_44_base_231_g5c411f69f&#34;,timestamp=&#34;2021-01-12 23:51:42&#34;,host=&#34;x86_64-w64-mingw32&#34;,lisp_name=&#34;SBCL&#34;,lisp_version=&#34;2.0.0&#34;,maxima_userdir=&#34;C:/Users/zmth1/maxima&#34;,maxima_tempdir=&#34;C:/Users/zmth1/AppData/Local/Temp&#34;,maxima_objdir=&#34;C:/Users/zmth1/maxima/binary/branch_5_44_base_231_g5c411f69f/sbcl/2_0_0&#34;,maxima_frontend=&#34;wxMaxima&#34;,maxima_frontend_version=&#34;20.12.2-DevelopmentSnapshot_MSW_OpenMP201511+Locks&#34;)build_info(version=&#34;branch_5_44_base_231_g5c411f69f&#34;,timestamp=&#34;2021-01-12 23:51:42&#34;,host=&#34;x86_64-w64-mingw32&#34;,lisp_name=&#34;SBCL&#34;,lisp_version=&#34;2.0.0&#34;,maxima_userdir=&#34;C:/Users/zmth1/maxima&#34;,maxima_tempdir=&#34;C:/Users/zmth1/AppData/Local/Temp&#34;,maxima_objdir=&#34;C:/Users/zmth1/maxima/binary/branch_5_44_base_231_g5c411f69f/sbcl/2_0_0&#34;,maxima_frontend=&#34;wxMaxima&#34;,maxima_frontend_version=&#34;20.12.2-DevelopmentSnapshot_MSW_OpenMP201511+Locks&#34;)(version=&#34;branch_5_44_base_231_g5c411f69f&#34;,timestamp=&#34;2021-01-12 23:51:42&#34;,host=&#34;x86_64-w64-mingw32&#34;,lisp_name=&#34;SBCL&#34;,lisp_version=&#34;2.0.0&#34;,maxima_userdir=&#34;C:/Users/zmth1/maxima&#34;,maxima_tempdir=&#34;C:/Users/zmth1/AppData/Local/Temp&#34;,maxima_objdir=&#34;C:/Users/zmth1/maxima/binary/branch_5_44_base_231_g5c411f69f/sbcl/2_0_0&#34;,maxima_frontend=&#34;wxMaxima&#34;,maxima_frontend_version=&#34;20.12.2-DevelopmentSnapshot_MSW_OpenMP201511+Locks&#34;)
-~~
+~~~
+~~~
 desolve(&#39;diff(f(x),x)+f(x),x)
-
+~~~
 get bogus irrelevant error message :
+~~~
 length: argument cannot be a symbol; found x
  -- an error. To debug this try: debugmode(true);
- 
+~~~ 
  when length has nothing to do with  it,irrelevant.

~~~~

- **status**: open --> accepted
- **assigned_to**: Leo Butler

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 18:03:50
Created by **[l_butler](https://sourceforge.net/u/l_butler/)** on 2021-12-08 13:20:18
Original: https://sourceforge.net/p/maxima/bugs/3899/#d8e2

---

The correct syntax is
~~~
desolve('diff(f(x),x),f(x));
~~~

The error message should be more informative. I will commit a fix for this shortly.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 18:03:54
Created by **[l_butler](https://sourceforge.net/u/l_butler/)** on 2021-12-08 14:43:05
Original: https://sourceforge.net/p/maxima/bugs/3899/#f76c

---

- **status**: accepted --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 18:03:57
Created by **[l_butler](https://sourceforge.net/u/l_butler/)** on 2021-12-08 14:43:05
Original: https://sourceforge.net/p/maxima/bugs/3899/#fc38

---

Fixed in commit [23db5eef5](https://sourceforge.net/p/maxima/code/ci/23db5eef5295b5d8e7d8042ef3491b6b7d7ec3aa/).
